ibis/backends/__init__.py

```diff
@@ -805,12 +812,7 @@ def __init__(self, *args, **kwargs):
         self._con_args: tuple[Any] = args
         self._con_kwargs: dict[str, Any] = kwargs
         self._can_reconnect: bool = True
-        # expression cache
-        self._query_cache = RefCountedCache(
-            populate=self._load_into_cache,
-            lookup=lambda name: self.table(name).op(),
-            finalize=self._clean_up_cached_table,
-        )
+        self._query_cache = RefCountedCache(weakref.proxy(self))
```

Comment:
`RefCountedCache` should probably be renamed to `Cache` or just anything that doesn't spell out its implementation, which isn't even correct anymore.